crates/nu-command/src/system/run_external.rs

```diff
@@ -418,13 +418,24 @@ impl ExternalCommand {
                             #[cfg(unix)]
                             {
                                 use nu_ansi_term::{Color, Style};
+                                use std::ffi::CStr;
                                 use std::os::unix::process::ExitStatusExt;
+
                                 if x.core_dumped() {
+                                    let cause = x
+                                        .signal()
+                                        .and_then(|sig| unsafe {
+                                            let sigstr_ptr = libc::strsignal(sig);
+                                            let sigstr: &'static CStr = CStr::from_ptr(sigstr_ptr);
+                                            sigstr.to_str().ok()
```

Comment:
Maybe worth adding a comment if we checked all safety assumptions.

https://man7.org/linux/man-pages/man3/strsignal.3.html

```suggestion
                                        .and_then(|sig| unsafe {
                                            // SAFETY: We should be the first to call `char * strsignal(int sig)`
                                            let sigstr_ptr = libc::strsignal(sig);
                                            let sigstr: &'static CStr = CStr::from_ptr(sigstr_ptr);
                                            sigstr.to_str().ok()
```